core/src/main/groovy/org/grails/gorm/graphql/types/output/AbstractObjectTypeBuilder.groovy

```diff
@@ -50,12 +52,17 @@ abstract class AbstractObjectTypeBuilder implements ObjectTypeBuilder {
 
     abstract GraphQLPropertyType getType()
 
-    protected GraphQLFieldDefinition.Builder buildField(GraphQLDomainProperty prop, String parentType) {
+    protected GraphQLFieldDefinition.Builder buildField(GraphQLDomainProperty prop, String parentType, MappingContext mapping) {
         GraphQLFieldDefinition.Builder field = newFieldDefinition()
                 .name(prop.name)
                 .deprecate(prop.deprecationReason)
                 .description(prop.description)
 
+        if(prop instanceof CustomGraphQLProperty) {
```

Comment:
This felt a little gross to me, but no other place stood out as a better alternative for actually building the property-level arguments.

Reply:
Agreed. You should use something like:

```
        if(prop instanceof Arguable) {
            List<GraphQLArgument> arguments = prop.getArguments(typeManager, mapping)
            if (!arguments.isEmpty()) {
                field.arguments(arguments)
            }
        }
```